ckanext/cfpb_extrafields/fanstatic/otherspecify.js

```diff
@@ -0,0 +1,14 @@
+$(function() {
+    $('select').change( function() { <!--select#type_of_entries-->
+                var value = $(this).val();
+                if (value == "__Other") {
+                    var custom = prompt( "Please enter custom value:" );
+                    if (!custom) return false;
+                        <!-- clear selections -->
+                        $(this).find($('option')).attr('selected',false)
+                        <!-- add and select new option -->
+                        $(this).prepend('<option value="' + custom + '" selected="selected">' + custom
```

Comment:
Be careful with this, just make sure your server-side code sanitizes the input before storing in the db.  Otherwise we could let users store arbitrary data in the db and perform SQL injection.